internal/bminventory/inventory.go

```diff
@@ -250,11 +250,27 @@ func (b *bareMetalInventory) updatePullSecret(pullSecret string, log logrus.Fiel
 func (b *bareMetalInventory) GetDiscoveryIgnition(ctx context.Context, params installer.GetDiscoveryIgnitionParams) middleware.Responder {
 	log := logutil.FromContext(ctx, b.log)
 
+	cluster, err := common.GetClusterFromDB(b.db, params.ClusterID, common.SkipEagerLoading)
+	if err != nil {
+		log.WithError(err).Errorf("failed to get cluster: %s", params.ClusterID)
+		return common.GenerateErrorResponder(err)
+	}
+
 	infraEnv, err := common.GetInfraEnvFromDB(b.db, params.ClusterID)
 	if err != nil {
+		log.WithError(err).Errorf("failed to get infra-env: %s", params.ClusterID)
 		return common.GenerateErrorResponder(err)
 	}
 
+	// update pull secret and proxy, since they could have been set during cluster update
```

Comment:
Would it not make more sense to set these in the infra env when they are set in the cluster?
Then we wouldn't have to worry about doing this all over again in the next endpoint that needs these values.

Reply:
We are in the middle of moving to V2. any new endpoints will use V2 infra env, which will be updated using dedicated API, without going through cluster. We can start adding code to ClusterUpdate API, but that function is confusing, and we are going to dissemble it anyway shortly